packages/web/src/components/ProjectRegistrationModal/RegistrationForm.tsx

```diff
@@ -0,0 +1,149 @@
+import { useFormik } from 'formik';
+import React, { useEffect, useState } from 'react';
+import * as Yup from 'yup';
+import {
+  Input,
+  FormControl,
+  VStack,
+  FormHelperText,
+  FormLabel,
+  Alert,
+  AlertDescription,
+  AlertIcon,
+  AlertTitle,
+  CloseButton,
+  Button,
+} from '@chakra-ui/react';
+
+interface ProjectProps {
+  name?: string;
+  description?: string;
+  tableNumber?: number;
+}
+
+export const RegistrationForm: React.FC = () => {
+  const [serverError, setServerError] = useState('');
+  const [project, setProject] = useState<ProjectProps>({});
+  const [validateWhileTyping, setValidateWhileTyping] = React.useState(false);
+
+  useEffect(() => {
+    const fetchProjects = async () => {
+      // todo default endpoint that user req.user to find the user's project/team
+      const res = await fetch('/api/project/');
+
+      try {
+        if (!res.ok) {
+          throw new Error();
+        }
+
+        const data = await res.json();
+        setProject(data);
+      } catch (err) {
+        // TODO print/log error message to show we were unable to pull their project data
+        setProject({});
+      }
+    };
+
+    void fetchProjects();
+  }, []);
+
+  const RegistrationSchema = Yup.object().shape({
+    name: Yup.string().required('Name is required'),
+    description: Yup.string(),
+    tableNumber: Yup.number(),
+  });
+
+  const formik = useFormik({
+    initialValues: {
+      name: project.name,
+      description: project.description,
+      tableNumber: project.tableNumber,
+    },
+    validationSchema: RegistrationSchema,
+    validateOnBlur: validateWhileTyping,
+    validateOnChange: validateWhileTyping,
+    async onSubmit(values) {
+      const res = await fetch('/api/project/register', {
+        method: 'POST',
+        body: JSON.stringify({
+          inputConfig: values,
+        }),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      });
+      if (!res.ok) {
+        setServerError('Something went wrong, please try again later...');
+      }
```

Comment:
On success, we probably either want to show a success modal or we want to invoke a callback that this component received as a prop

@johnkahn any preference?